app/main/views/questionnaire.py

```diff
@@ -58,6 +58,9 @@ def post_questionnaire(eq_id, form_type, period_id, collection_id, location):
         return render_page(location, False)
 
     next_location = g.questionnaire_manager.get_current_location()
+    if next_location == location:
```

Comment:
Thanks @benjefferies  and @jonnyshaw89. I guess this makes sense if we're planning on getting rid of the User Action pattern altogether. Some actions are never used, so perhaps dedicated end points for actions, possibly an actions Blueprint instead?